run_travis_task.py

```diff
@@ -0,0 +1,118 @@
+#!/usr/bin/env python3
+# -*- encoding: utf-8 -*-
+"""
+Because we have a lot of Travis jobs, we try to avoid running unnecessary
+tests and deploys.  This speeds up our builds, minimises deployment churn,
+and reduces contention on our Travis infrastructure.
+
+How we decide what to build:
+
+                  | Relevant changes | No relevant changes
+   ---------------+------------------+---------------------
+    pull request  | Run tests        | Don't run tests
+                  | Don't deploy     | Don't deploy
+   ---------------+------------------+---------------------
+    master        | Run tests        | Run tests
+                  | Deploy to AWS    | Don't deploy
+
+We always run tests on master so we get consistent build results, that's
+less important on master where results are transient.
```

Comment:
I think you meant less important on pull requests?

Reply:
Oops. Yes, will fix.